packages/azure-kusto-data/src/connectionBuilder.browser.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { DeviceCodeInfo } from "@azure/identity";
+import KustoConnectionStringBuilderBase from "./connectionBuilderBase";
+
+export class KustoConnectionStringBuilder extends KustoConnectionStringBuilderBase {
+    static readonly DefaultDatabaseName = "NetDefaultDB";
+    static readonly SecretReplacement = "****";
+    // eslint-disable-next-line no-console
+
+    // TODO delete?
+    static withAadUserPasswordAuthentication(connectionString: string, userId: string, password: string, authorityId?: string) {
+        if (userId.trim().length === 0) throw new Error("Invalid user");
+        if (password.trim().length === 0) throw new Error("Invalid password");
+
+        const kcsb = new KustoConnectionStringBuilder(connectionString);
+        kcsb.aadFederatedSecurity = true;
+        kcsb.aadUserId = userId;
```

Comment:
is this taking a user's AAD password? We wouldn't want an app to collect this in the browser, surely

Reply:
deleting